app/uat_actions/uat_actions/tube_submission.rb

```diff
@@ -0,0 +1,153 @@
+# frozen_string_literal: true
+
+# This UAT Action will generates a basic submission for tubes. Initially, it
+# has been designed for generating scRNA Core Donor Pooling and cDNA Prep
+# submissions on LRC Bank Seq/Spare tubes.
+class UatActions::TubeSubmission < UatActions
+  self.title = 'Tube submission'
+  self.description = 'Generates a basic submission for tubes.'
+  self.category = :setup_and_test
+
+  form_field :submission_template_name,
+             :select,
+             label: 'Submission Template',
+             help: 'Select the submission template to use.',
+             select_options: -> { compatible_submission_templates }
+
+  form_field :tube_barcodes,
+             :text_area,
+             label: 'Tube barcodes',
+             help: 'Add the tubes which will form part of your submission.'
+
+  form_field :library_type_name,
+             :select,
+             label: 'Library Type',
+             help:
+               'Select the library type to use when creating the requests. ' \
+                 'Leave blank to automatically use the first library type found. ' \
+                 'Useful where the same request type has multiple library types.',
+             select_options: -> { LibraryType.alphabetical.pluck(:name) },
+             options: {
+               include_blank: 'Using default library type...'
+             }
+
+  validates :submission_template, presence: { message: 'could not be found' }
+
+  # Returns a default copy of the UatAction which will be used to fill in the form
+  #
+  # @return [UatActions::TestSubmission] A default object for rendering a form
+  def self.default
+    new
+  end
+
+  # Returns the submission templates which are compatible with the UAT action.
+  # These are submission templates which have a tube as an input asset type.
+  #
+  # @return [Array<String>] The names of the compatible submission templates
+  def self.compatible_submission_templates
+    SubmissionTemplate
+      .visible
+      .each_with_object([]) do |submission_template, compatible|
+        next unless submission_template.input_asset_type.constantize < Tube
```

Comment:
I didn't you could use `<` as a comparison operator like this. 
What if the `input_asset_type` is a Tube (rather than inherits from it), is that a possibility?

Reply:
Oh, it should do <= to include Tube as well. Good catch.

[1] pry(main)> Tube <= Tube
=> true
[2] pry(main)> SampleTube <= Tube
=> true